src/kirin/dialects/ilist/stmts.py

```diff
@@ -35,7 +35,10 @@ def __init__(
         elem_type: types.TypeAttribute | None = None,
     ) -> None:
         if not elem_type:
-            elem_type = types.Any
+            if not values:
+                elem_type = types.Any
+            else:
+                elem_type = values[0].type
```

Comment:
this is not correct? the type should be union of all elements being used to construct the list.

Reply:
Oh, my bad. Thanks for fixing it!